laurat007/assignment3/Trie.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/**
+ *
+ * @author Laura
+ *  Implemented a Trie structure to build a dictionary.
+ *  The Trie works like this: when a word is added, the trie instantiates 
+ * its nth child (n = first letter of the word - 'A' in ASCII codes), so it 
+ * goes increasing the height of the trie, simultaneously with the length 
+ * of the word. If the nth child is already instantiated, the algorithm goes
+ * to the next step.
+ *  
+ */
+ 
+public class Trie {
+     
+    static final int MAX_LETTERS = 26;
+    
+    class TrieNode
+    {
+        TrieNode[] children = new TrieNode[MAX_LETTERS];
+      
+        boolean isEndOfWord;
+         
+        TrieNode(){
+            isEndOfWord = false;
+            for (int i = 0; i < MAX_LETTERS; i++)
+                children[i] = null;
```

Comment:
You don't need to instantiate Objects to null, this is the default value java assigns so this is an unnecessary loop

Reply:
right. done.